aten/src/ATen/core/ivalue_inl.h

```diff
@@ -522,6 +528,25 @@ c10::optional<T> generic_to(
   return std::move(ivalue).to<T>();
 }
 
+template <typename Tuple, std::size_t... I>
+Tuple generic_to_tuple_impl(
+    const std::vector<IValue>& t,
+    c10::guts::index_sequence<I...>) {
+  return std::make_tuple(
+      t[I].to<typename std::tuple_element<I, Tuple>::type>()...);
+}
```

Comment:
Nit: avoid dumping helper functions like this in global namespace; put it in `namespace detail { ... }` or similar